cmd/jsonnet.cpp

```diff
@@ -673,38 +685,76 @@ int main(int argc, const char **argv)
+                        }
+
+                        std::string input;
+                        if (!read_input_contents(inputFile, &input)) {
+                            jsonnet_destroy(vm);
+                            return EXIT_FAILURE;
+                        }
+
+                        output = jsonnet_fmt_snippet(vm, inputFile.c_str(), input.c_str(), &error);
+
+                        if (error) {
+                            std::cerr << output;
+                            std::cerr.flush();
+                            jsonnet_realloc(vm, output, 0);
+                            jsonnet_destroy(vm);
+                            return EXIT_FAILURE;
+                        }
+
+                        if (config.fmtTest) {
+                            // Check the output matches the input.
+                            bool ok = output == input;
+                            jsonnet_realloc(vm, output, 0);
+                            jsonnet_destroy(vm);
+                            return ok ? EXIT_SUCCESS : 2;
+                        } else {
+                            // Write output Jsonnet.
+                            bool successful = write_output_file(output, output_file);
+                            jsonnet_realloc(vm, output, 0);
+                            if (!successful) {
+                                jsonnet_destroy(vm);
+                                return EXIT_FAILURE;
+                            }
+                        }
                     }
-                    if (config.filenameIsCode) {
-                        std::cerr << "ERROR: Cannot use --in-place with --exec" << std::endl;
+                } else {
+                    assert(config.inputFiles.size() == 1);
+                    // Read input file.
+                    std::string input;
+                    if (!read_input(&config, 0, &input)) {
                         jsonnet_destroy(vm);
                         return EXIT_FAILURE;
                     }
-                    output_file = config.inputFile;
-                }
 
-                output = jsonnet_fmt_snippet(vm, config.inputFile.c_str(), input.c_str(), &error);
+                    output = jsonnet_fmt_snippet(vm, config.inputFiles[0].c_str(), input.c_str(), &error);
 
-                if (error) {
-                    std::cerr << output;
-                    std::cerr.flush();
-                    jsonnet_realloc(vm, output, 0);
-                    jsonnet_destroy(vm);
-                    return EXIT_FAILURE;
-                }
-
-                if (config.fmtTest) {
-                    // Check the output matches the input.
-                    bool ok = output == input;
-                    jsonnet_realloc(vm, output, 0);
-                    jsonnet_destroy(vm);
-                    return ok ? EXIT_SUCCESS : 2;
+                    if (error) {
+                        std::cerr << output;
+                        std::cerr.flush();
```

Comment:
STDERR is unbuffered, I don't think this is needed: https://linux.die.net/man/3/stderr